rpc_server.go

```diff
@@ -196,23 +222,35 @@ type SizeObject struct {
 
 // ConfigurePulseLengths is the RPC-callable service to change pulse record sizes.
 func (s *SourceControl) ConfigurePulseLengths(sizes SizeObject, reply *bool) error {
+	*reply = false
 	log.Printf("ConfigurePulseLengths: %d samples (%d pre)\n", sizes.Nsamp, sizes.Npre)
 	if !s.isSourceActive {
 		return fmt.Errorf("No source is active")
 	}
-	if s.ActiveSource.ComputeWritingState().Active && (s.status.Npresamp != sizes.Npre || s.status.Nsamples != sizes.Nsamp) {
+	if s.status.Npresamp == sizes.Npre && s.status.Nsamples == sizes.Nsamp {
+		return nil // No change requested
+	}
+	if s.ActiveSource.ComputeWritingState().Active {
 		return fmt.Errorf("Stop writing before changing record lengths")
 	}
-	err := s.ActiveSource.ConfigurePulseLengths(sizes.Nsamp, sizes.Npre)
+
+	f := func() {
+		err := s.ActiveSource.ConfigurePulseLengths(sizes.Nsamp, sizes.Npre)
+		if err == nil {
+			s.status.Npresamp = sizes.Npre
+			s.status.Nsamples = sizes.Nsamp
+		}
+		s.broadcastStatus()
```

Comment:
No particular reason. I suspect we want it to be unconditional in both cases, so that a DC instance that asks for a change to Nsamp or trigger conditions will be notified that its GUI data is incorrect by the Status or Trigger message. So I'll change line 170 to be unconditional `broadcastTriggerState()`.